WordPress/Classes/ViewRelated/NUX/LoginSocialErrorViewController.swift

```diff
@@ -0,0 +1,153 @@
+import Foundation
+import Gridicons
+
+class LoginSocialErrorViewController: UITableViewController {
+    fileprivate var errorTitle: String
+    fileprivate var errorDescription: String
+
+    fileprivate enum Sections: Int {
+        case titleAndDescription = 0
+        case buttons = 1
+
+        static var count: Int {
+            return buttons.rawValue + 1
+        }
+    }
+
+    fileprivate enum Buttons: Int {
+        case tryEmail = 0
+        case tryAddress = 1
+        case signup = 2
+    }
+
+    init(title: String, description: String) {
+        errorTitle = title
+        errorDescription = description
+
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        errorTitle = aDecoder.value(forKey: "errorTitle") as? String ?? ""
+        errorDescription = aDecoder.value(forKey: "errorDescription") as? String ?? ""
+
+        super.init(coder: aDecoder)
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        tableView.estimatedRowHeight = 100.0
+
+        view.backgroundColor = WPStyleGuide.greyLighten30()
+    }
+
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        guard indexPath.section == Sections.buttons.rawValue else {
+            return
+        }
+
+        let controllerKey: String
+        switch indexPath.row {
+        case Buttons.tryEmail.rawValue:
+            controllerKey = "emailEntry"
+        case Buttons.tryAddress.rawValue:
+            controllerKey = "siteAddress"
+        case Buttons.signup.rawValue:
+            fallthrough
+        default:
+            controllerKey = "SignupViewController"
+        }
+
+        let storyboard = UIStoryboard(name: "Login", bundle: nil)
+        if let controller = storyboard.instantiateViewController(withIdentifier: controllerKey) as? NUXAbstractViewController {
+            navigationController?.setViewControllers([controller], animated: true)
```

Comment:
I'm not sure about this.  Summoning @folletto.  
When I first tapped the email option to retry another view controller appeared to be pushed onto the stack. My expectation, since the error screen was pushed onto the nav stack, was that I'd go back a screen.  But then there's no back button so that's a bit odd. 
When I tried the option to sign up I was trapped with no way to get back to the login flow.
This is similar for the site address screen... there is no way to return to the email screen. 

Given the scenarios above I think it would make sense to present the error screen as a modal and maybe let the presenting controller decide what action to take via a delegate method.  Thoughts?

Reply:
Ya, I could see model presentation + delegate as a nice UX.

I did it this way because of the navigation bar being part of these screens. I'd also like to hear @folletto's take 😀

Reply:
I guess I need to ask a screenshot of the 2+ screens you're discussing about. It's a bit too theoretical in writing :D

Reply:
@folletto here's a demonstration: http://d.pr/v/bX1dEZ

Producing that video made it clear that @aerych has a point. A bit of direction on how best to alter it please 😀

Reply:
The "← Back" action should be present there, behaving exactly like the user just opened that screen in a normal flow of actions, which means that the "Back":

* On the site address screen, it should go back to the email screen (where normally the user would select to enter site address).
* On the email screen, it should go back to the intro screen.

This is because these actions there act as "shortcuts" into a specific part of the flow.

Reply:
> The "← Back" action should be present there, behaving exactly like the user just opened that screen in a normal flow of actions

I have concerns about this.  In the video example Nate shared new screens are sliding in from the right and the error screen is transitioned away by sliding to the left... a typical navigation experience.  This implies that tapping back will take the user to the error screen, but if I understand correctly, we're suggesting that we break this expectation.   So the experience would be the error screen slides in, the user taps an option on the error screen, a new screen slides in. The user taps the back button... but rather than seeing the error screen they see something different.  I expect this would be confusing.

I think it would be better for the error screen presented modally so it slides up from the bottom of the screen and dismisses by sliding down. This removes any expectation that it is something the user would see by tapping a back button.

Reply:
Yep — the sliding motion should be different too with this model.

If we want to do it modal we need to go back to the design and rework both iOS and Android at this point.

Reply:
Caught up with @folletto in chat.  To summarize ... we're talking about the same thing but speaking in different terms. :D  So let's go ahead and present the error screen "modally" (such an ambiguous term), full screen, sliding up from the bottom, etc. so we avoid any weirdness with user expectations and the back button. :)